### PR TITLE
[Parser] Parse explicit exports

### PR DESCRIPTION
**Pull request by tlively**

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #6150** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6174** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6174?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #6179** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6179?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6179?utm_source=stack-comment).

File: src/parser/contexts.h [hunk 1]
```diff
@@ -684,6 +687,31 @@ struct ParseDeclsCtx : NullTypeParserCtx, NullInstrParserCtx {
...
+
+  Result<> addGlobalExport(Index pos, GlobalIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
+
+  Result<> addTagExport(Index pos, TagIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
```
Comment (kripken) [inline src/parser/contexts.h:714]: Can we use a template over these 5 on the second param?

Comment (tlively) [reply src/parser/contexts.h:714]: Hmm yeah I'll try that out.

Comment (tlively) [reply src/parser/contexts.h:714]: Ok, there's only a single `addExport` method on each context now.

**Review by kripken — APPROVED** [review]

**Pull request merged**